Neos.Flow/Classes/Cache/AnnotationsCacheFlusher.php

```diff
@@ -0,0 +1,85 @@
+<?php
+declare(strict_types=1);
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+namespace Neos\Flow\Cache;
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Reflection\ReflectionService;
+use Neos\Flow\Log\Utility\LogEnvironment;
+use Psr\Log\LoggerInterface;
+
+final class AnnotationsCacheFlusher
+{
+    /**
+     * @Flow\Inject
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @Flow\Inject
+     * @var ConfigurationManager
+     */
+    protected $configurationManager;
+
+    /**
+     * @Flow\Inject
+     * @var CacheManager
+     */
+    protected $cacheManager;
+
+    /**
+     * The ReflectionService is only needed during compile time for handling flushConfigurationCachesByCompiledClass
+     *
+     * @Flow\Inject
+     * @var ReflectionService
+     */
+    protected $reflectionService;
+
+    /**
+     * A slot that flushes caches as needed if classes with specific annotations have changed
+     *
+     * @param array<string> $classNames The full class names of the classes that got compiled
+     * @return void
+     */
+    public function flushConfigurationCachesByCompiledClass(array $classNames): void
+    {
+        $caches = [
+            Neos\Flow\Annotations\Route::class => ['Flow_Mvc_Routing_Route', 'Flow_Mvc_Routing_Resolve']
```

Comment:
```suggestion
            Neos\Flow\Annotations\Route::class => ['Flow_Mvc_Routing_Route', 'Flow_Mvc_Routing_Resolve'],
            Neos\Flow\Annotations\Privilege::class => ['Flow_Security_Policy']
```

Reply:
I didn't have the cance to test this (I will tomorrow) but I was wondering: Is flushing the routing/policy caches enough? since the annotations affect the _configuration_ too..?

Reply:
The namespace `Neos\Flow\Annotations\Route` doesn't exist

Reply:
> The namespace `Neos\Flow\Annotations\Route` doesn't exist

Not an issue effectively, because as you know, you can do `My\Madeup\Namespace\Class::class` and it will just resolve to the proper class string without checking if the class actually exists.
But as said, maybe we do want to add those mappings in the specific annotation PRs only.

Reply:
It was missing the trailing backslash, _that's_ what I meant :)